### DISCUSSION TITLE
Pre-build of Pocketbase for Android and iOS

### DISCUSSION THREAD

**Discussion by iampopal** [pocketbase/pocketbase#3044] [Q&A]
We are planning to develop a saas over Pocketbase, 
In our SAAS the Pocketbase server will be not running in the cloud, it will be running locally on users' computers. 

Like: One computer server and other computers connect to that server. 

As our Pocketbase server will be running locally, for security reasons our server need to backup its database to the cloud. 

After getting back up to the cloud, as we have a mobile app too, Users will download the database data.db backup to their phone and will like to open it for reviewing changes. 

 So here is a short list of why we need local backup: 
 1) Employees run Pocketbase locally on their computers and backup data.db at the end of the day to the cloud. 
 2) Owners download data.db in mobile app (Android, iOS) and want to open their database locally.
 
 We are going offline first and then backup because of in the area that we implement our saas not has very active internet connection all the time.

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand what exactly is the request here.

If you are asking for Android and iOS binaries then this can't be done at the moment without complicating the release flow because we use the pure Go SQLite port by default ([modernc.org/sqlite](https://pkg.go.dev/modernc.org/sqlite)) and based on the ["Supported platforms and architectures" list](https://pkg.go.dev/modernc.org/sqlite#hdr-Supported_platforms_and_architectures), Android and iOS is not supported as GOOS target (_I'm not sure if the existing `linux` and `darwin` targets work and if not I have no clue about the differences between the `android`/`ios`_).

If you want to build PocketBase on Android/iOS, you most likely will have to use the CGO  driver ([mattn/go-sqlite3](https://github.com/mattn/go-sqlite3)) which is enabled when `CGO_ENABLED=1` is set. But keep in mind that when using CGO the executable is no longer statically linked by default and you'll need to build your application in a container OR on the device itself to ensure that the correct env dependencies are linked:

0. [Install Go 1.18+](https://go.dev/doc/install) (if you haven't already)
1. Clone/download the repo
2. Navigate to examples/base
3. Run `GOOS=android GOARCH=arm64 CGO_ENABLED=1 go build` (https://go.dev/doc/install/source#environment)
4. Start the created executable by running ./base serve.

**Reply by iampopal** [reply]
Thank you, 

Yes, we need binary for Android and iOS we are not Go developers and didn't work with Go language yet so wondering if possible to get the binary of Android and iOS as a package for Dart as we include it automatically installs Pocketbase binary and we able to open Pocketbase data.db database in the Android/iOS app.  

Like this package: https://pub.dev/packages/sqlite3_flutter_libs

Automatically add flutter libs to the repository and we can open SQLite database.db file in any platform

**Comment by lavvy** [comment]
Any information so far ?
